gammapy/maps/geom.py

```diff
@@ -631,29 +631,58 @@ def energy_mask(self, energy_min=None, energy_max=None, round_to_edge=False):
         ----------
         energy_min, energy_max : `~astropy.units.Quantity`
             Energy range.
+        round_to_edge: bool, optional
+            Wether to round `energy_min` and `energy_max` to the closest axis bin value.
+            See `~gammapy.maps.MapAxis.round`. Default is False.
 
         Returns
         -------
         mask : `~gammapy.maps.Map`
             Map containing the energy mask. The geometry of the map
             is the same as the geometry of the instance which called this method.
         """
+        return self.create_mask(
+            "energy",
+            edge_min=energy_min,
+            edge_max=energy_max,
+            round_to_edge=round_to_edge,
+        )
+
+    def create_mask(self, axis_name, edge_min=None, edge_max=None, round_to_edge=False):
+        """Create a mask over a given axis.
+
+        Bins must be fully contained to be included in the mask.
+
+        Parameters
+        ----------
+        edge_min, edge_max : `~astropy.units.Quantity` or float
+            Bounds of the mask. If the axis used for the mask has a unit, passing `~astropy.units.Quantity` is mandatory.
+        round_to_edge: bool, optional
+            Wether to round `energy_min` and `energy_max` to the closest axis bin value.
+            See `~gammapy.maps.MapAxis.round`. Default is False.
+
+        Returns
+        -------
+        mask : `~gammapy.maps.Map`
+            Map containing the mask. The geometry of the map is the same as the initial geometry.
+        """
         from . import Map
 
-        # get energy axes and values
-        energy_axis = self.axes["energy"]
+        axes_names = self.axes_names
 
-        if round_to_edge:
-            energy_min, energy_max = energy_axis.round([energy_min, energy_max])
+        axis = self.axes[axis_name]
 
-        # TODO: make this more general
-        shape = (-1, 1) if self.is_hpx else (-1, 1, 1)
```

Comment:
Can you completely ignore the special case for hpx here?

Reply:
Yes I think we can.